src/pages/workspace/WorkspacesListPage.tsx

```diff
@@ -363,7 +363,7 @@ function WorkspacesListPage({policies, allPolicyMembers, reimbursementAccount, r
                 <HeaderWithBackButton
                     title={translate('common.workspaces')}
                     shouldShowBackButton={isSmallScreenWidth}
-                    onBackButtonPress={() => Navigation.goBack()}
+                    onBackButtonPress={() => Navigation.goBack(ROUTES.ALL_SETTINGS)}
```

Comment:
What does this fix? This causes regression of wrong navigation animation

Repro step:
1. Visit https://dev.new.expensify.com:8082/settings/workspaces directly on mWeb
2. Click back button

main:

https://github.com/Expensify/App/assets/108292595/679c3c84-ac40-4638-a8fc-04db1519b37b


this branch:

https://github.com/Expensify/App/assets/108292595/a6ddf225-fc3a-4fa7-901a-256dc8a97e41